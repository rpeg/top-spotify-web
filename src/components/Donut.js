import React from 'react';
import PieChart from 'react-minimal-pie-chart';
import Palette from 'google-palette';
import PropTypes from 'prop-types';

import { makeSortedFrequencyArr } from '../lib/frequency';

const Donut = ({ items }) => {
  const sortedItems = makeSortedFrequencyArr(items);

  const colors = Palette('tol-rainbow', sortedItems.length).map((c) => `#${c}`);

  const donutData = sortedItems.map((item, i) => ({
    color: colors[i],
    title: item.name,
    value: item.freq,
  }));

  return (
    <div>
      <PieChart
        className="minimal-pie"
        animate
        animationDuration={750}
        animationEasing="ease-out"
        cx={50}
        cy={50}
        data={donutData}
        label={({ data, dataIndex }) => data[dataIndex].title}
        labelPosition={60}
        labelStyle={{
          fontFamily: 'Karla, sans-serif',
          fontSize: '6px',
        }}
        lengthAngle={360}
        lineWidth={20}
        onClick={undefined}
        onMouseOut={undefined}
        onMouseOver={undefined}
        paddingAngle={0}
        radius={50}
        ratio={1}
        rounded={false}
        startAngle={0}
      />
    </div>
  );
};

Donut.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Donut;
