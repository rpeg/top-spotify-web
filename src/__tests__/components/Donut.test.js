import React from 'react';
import { shallow } from 'enzyme';

import Donut from '../../components/Donut';

describe('Donut', () => {
  const items = ['a', 'b'];

  it('should not log errors in console', () => {
    const wrapper = shallow(<Donut items={items} />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Donut items={items} />);
    expect(wrapper).toMatchSnapshot();
  });
});
