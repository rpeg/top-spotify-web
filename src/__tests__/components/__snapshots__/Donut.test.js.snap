// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donut renders correctly 1`] = `
<div>
  <ReactMinimalPieChart
    animate={true}
    animationDuration={750}
    animationEasing="ease-out"
    className="minimal-pie"
    cx={50}
    cy={50}
    data={
      Array [
        Object {
          "color": "#781c81",
          "title": "b",
          "value": 1,
        },
        Object {
          "color": "#d92120",
          "title": "a",
          "value": 1,
        },
      ]
    }
    label={[Function]}
    labelPosition={60}
    labelStyle={
      Object {
        "fontFamily": "Karla, sans-serif",
        "fontSize": "6px",
      }
    }
    lengthAngle={360}
    lineWidth={20}
    paddingAngle={0}
    radius={50}
    ratio={1}
    rounded={false}
    startAngle={0}
    viewBoxSize={
      Array [
        100,
        100,
      ]
    }
  />
</div>
`;
